docs/CNS.md

```diff
@@ -0,0 +1,166 @@
+
+If no suitable compiler is found, you will get an error message:
+
+```
+Error: no suitable compiler (ifort, pgf95, ifort_i4, ifort_mp or pgf95_mp) found
+```
+
+In this case, you will have to install a compiler. For more information, check the [CNS Installation page](http://cns-online.org/v1.3/installation/frame.html).
+One Fortran compiler that is easy (and free) to install is `gfortran`. It is part of the GNU Compiler Collection (GCC), which can be installed with one command if you have homebrew:
+
+```bash
+brew install gcc
+```
+
+Alternatively, gfortran can be downloaded as a [separate binary](https://gcc.gnu.org/wiki/GFortranBinaries).
+
+#### Write a Makefile header
+
+If a suitable compiler is installed but no corresponding `Makefile.header` is found, `make install` will give an error as well. For example (with gfortran):
+
+```
+Error: Makefile template for compiler gfortran is not available
+```
+
+Makefile headers for each supported system-compiler combination are provided in `instlib/machine/supported/` (for unsupported systems, see `instlib/machine/unsupported/`). If you already tried `make install`, the appropriate directory for your system (eg. `mac-intel-darwin`) will have been copied to the main `cns_solve_1.3` directory.
+If this directory does not contain the desired file (eg. `Makefile.header.2.gfortran`), create one manually.
+
+For example, for `gfortran`, create a new file named `Makefile.header.6.gfortran` inside the directory containing Makefile headers for your system. Save the following lines in this file:
+
+```
+# fortran options
+F77 = gfortran
+F77STD = -fdefault-integer-8 -w -fallow-argument-mismatch
+F77OPT = -O3 $(CNS_MALIGN_I86) -funroll-loops -ffast-math -march=native -mtune=native -static
+F77FLAGS = $(F77STD) $(F77OPT) $(EXT_F77FLAGS) $(F77BUG)
+
+# C options
+CC = gcc
+CPP = g++
+CCFLAGS = -O -DINTEGER='long int' -DCNS_ARCH_TYPE_$(CNS_ARCH_TYPE) $(EXT_CCFLAGS)
+
+# link options
+LD = gfortran
+LDFLAGS = -w $(EXT_LDFLAGS)
+```
+
+Note the `-fallow-argument-mismatch` flag. From gfortran version 10 onward, type mismatches by default give an error rather than a warning. This flag returns warnings instead, so that CNS compilation can successfully complete. If a `Makefile.header.x.gfortran` is provided for your system but compilation fails with (many) `Error: Type mismatch`, you may need to add this flag to the provided file.
+
+Then try [`make install`](#Start-compilation) again, as described above.
+
+## 4 Source CNS
+
+Lastly, CNS needs to be loaded in the current shell before it can be used.
+
+For csh/tcsh:
+
+```bash
+source cns_solve_env
+```
+
+For sh/bash:
+
+```bash
+source .cns_solve_env_sh
+```
+
+To avoid having to repeat this step each time you open a terminal window to run CNS/HADDOCK, you may want to add the relevant `source` line to your `~/.cshrc`, `~/.tcshrc`, `~/.bashrc`, or `~/.bash_profile`. When doing this, make sure to use the full path instead of only the file name.
```

Comment:
I think this is not needed to run HADDOCK. At least, I don't have it in my `.bashrc`. It doesn't hurt, though :relaxed: